### PR TITLE
feat: Improve debug logging for restore and save steps

### PR DESCRIPTION
**Pull request by andersfischernielsen**
This PR will:
- add debug logging in order to make debugging cache issues during CI runs easier 
- split existing logs for readability

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -82,6 +85,7 @@ export async function runAction(action: () => Promise<unknown>) {
  * @returns treeHash
  */
 export async function getTreeHashForCommitHash(commit: string) {
+    core.debug(`Executing: 'git rev-parse', [\`${commit}:\`]`)
```
Comment (pekala) [inline src/utils.ts:88]: The `@actions/exec` already prints the command that is executed (you can see it here: 
https://user-images.githubusercontent.com/4643658/180269047-417226dd-ce8f-41a6-92ee-e6ed7d279cb6.png ) 

We could consider passing the `silent` option to `exec`, if we want these to only show in debug mode

File: src/restore.ts [hunk 2]
```diff
@@ -15,16 +15,19 @@ runAction(async () => {
...
-    // "post" run of the action (save.ts)
+    // Saving key and hash in "state" which can be retrieved by the "post" run of the action (save.ts)
     // https://github.com/actions/toolkit/tree/daf8bb00606d37ee2431d9b1596b88513dcf9c59/packages/core#action-state
-    core.saveState('key', output.key)
-    core.saveState('hash', output.treeHash)
-
-    core.setOutput('processed', output.processed)
-    core.setOutput('hash', output.treeHash)
+    core.saveState('key', key)
+    core.debug(`Saved state: ${key}`)
```
Comment (pekala) [inline src/restore.ts:23]: GitHub has built-in debug logs for setting state and outputs

**Review by pekala — COMMENTED** [review]
The `exec` stuff is logged already - not sure if we need it again. For setting state and output - GitHub already does that (see e.g. here https://github.com/pleo-io/product-web/runs/7466151775?check_suite_focus=true#step:5:52)

**Comment by andersfischernielsen** [comment]
Closing due to feedback.

**Pull request closed without merging**